packages/utils/schema/user-profile.graphql

```diff
@@ -0,0 +1,111 @@
+type ProfileName
+  @createModel(accountRelation: SINGLE, description: "Your name") {
+  name: String! @string(maxLength: 150)
+}
+
+type ProfileAvatar
+  @createModel(
+    accountRelation: SINGLE
+    description: "An image and optional fallback images representing your avatar"
+  ) {
+  original: ImageMetadata!
+  alternatives: [ImageMetadata] @list(maxLength: 3)
+}
+
+type ProfileDescription
+  @createModel(
+    accountRelation: SINGLE
+    description: "A short profile bio / blurb / description"
+  ) {
+  description: String! @string(maxLength: 420)
+}
+
+type ProfileEmoji
+  @createModel(accountRelation: SINGLE, description: "Your spirit emoji") {
+  emoji: String! @string(maxLength: 2)
+}
+
+type ProfileHeaderBackground
+  @createModel(
+    accountRelation: SINGLE
+    description: "An image (3:1 ratio) and optional fallback images to be used as a header background"
+  ) {
+  original: ImageMetadata!
+  alternatives: [ImageMetadata] @list(maxLength: 3)
+}
+
+type ProfileURL
+  @createModel(
+    accountRelation: SINGLE
+    description: "Your home on the internet"
+  ) {
+  url: String! @string(maxLength: 240)
+}
+
+type ProfileHomeLocation
+  @createModel(accountRelation: SINGLE, description: "Where you live") {
+  homeLocation: String! @string(maxLength: 140)
+}
+
+type ProfileUsername
+  @createModel(
+    accountRelation: SINGLE
+    description: "A representation of your identity that can be used in an interface, typically as part of a URL"
+  ) {
+  # pattern: "^[a-z0-9-_]+$"
```

Comment:
We'll still need to have pretty strict requirements for how this value is structured.

I like the idea of allowing capital letters, but doing the uniqueness check based on a purely lowercase string.

I think the character set should be restricted to whatever is permissible as part of a subdomain name. Limited to a single position though, so no `.`s.

If the name submitted contains a dot, it is derefenced using ENS.

Reply:
Patterns are unfortunately not supported yet, but they're on the Ceramic team's roadmp